src/state/features/userSlice.ts

```diff
@@ -1,37 +1,75 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit'
 
-type userInfoType = {
+interface classesType {
+  cid: string
+  cType: boolean
+}
+
+interface userInfoType {
+  _id: string
   name: string
   email: string
   img?: string
   isLoggedIn: boolean
+  isAdmin: boolean
+  classes: classesType[]
+  made: string[]
+  madeOptions: string[]
+  solved: string[]
+}
+
+interface userEnrollType {
+  cid: string
+  cType: boolean
 }
 
 const initialState: userInfoType = {
+  _id: '',
   name: '',
   email: '',
   img: '',
   isLoggedIn: false,
+  isAdmin: false,
+  classes: [],
+  made: [],
+  madeOptions: [],
+  solved: [],
 }
 
 export const userSlice = createSlice({
   name: 'userInfo',
   initialState,
   reducers: {
     login: (state: userInfoType, action: PayloadAction<userInfoType>) => {
+      state._id = action.payload._id
       state.name = action.payload.name
       state.email = action.payload.email
       state.img = action.payload.img
       state.isLoggedIn = action.payload.isLoggedIn
+      ;(state.isAdmin = action.payload.isAdmin),
```

Comment:
is there any reason you put brackets?
```suggestion
      state.isAdmin = action.payload.isAdmin
```

Reply:
> is there any reason you put brackets?

I am not sure why prettier added that. Probably because there was a comma at the end of the previous line. I have adjusted it now :)